src/dialog.js

```diff
@@ -449,6 +457,134 @@ genotet.dialog.upload_ = function() {
 };
 
 /**
+ * Creates a dialog for signing up.
+ * @private
+ */
+genotet.dialog.signUp_ = function() {
+  var modal = $('#dialog');
+  modal.find('.modal-content').load(genotet.dialog.TEMPLATES_.signUp,
+    function() {
+      modal.modal();
+      var email = modal.find('#email');
+      var username = modal.find('#username');
+      var password = modal.find('#password');
+      var confirmPassword = modal.find('#confirm-password');
+      var btnSignUp = modal.find('#btn-sign-up').prop('disabled', true);
+      var inputValid = false;
+
+      var uploadReady = function() {
+        // Validate the input information.
+        inputValid = genotet.utils.validateEmail(email.val()) &&
+          genotet.utils.validateUsername(username.val()) &&
+          genotet.utils.validatePassword(password.val());
+
+        // Checks if all required fields are filled.
+        return email.val() && username.val() && password.val() &&
+          (password.val() == confirmPassword.val() && inputValid);
+      };
+      confirmPassword.on('input', function() {
+        btnSignUp.prop('disabled', !uploadReady());
+      });
+      btnSignUp.click(function() {
+        var userInfo = {
+          type: 'sign-up',
+          email: email.val(),
+          username: username.val(),
+          password: CryptoJS.SHA256(password.val()).toString(),
+          confirmed: false
+        };
+
+        $.ajax({
+          url: genotet.data.userURL,
+          type: 'POST',
+          data: userInfo,
+          dataType: 'json'
+        }).done(function(data) {
+            if (!data.success) {
```

Comment:
I've already discussed the reason in other reply.

On Saturday, March 19, 2016, Liana Yang <notifications@github.com> wrote:

> In src/dialog.js
> <https://github.com/ViDA-NYU/genotet/pull/145#discussion_r56754303>:
>
> > +      btnSignUp.click(function() {
> > +        var userInfo = {
> > +          type: 'sign-up',
> > +          email: email.val(),
> > +          username: username.val(),
> > +          password: CryptoJS.SHA256(password.val()).toString(),
> > +          confirmed: false
> > +        };
> > +
> > +        $.ajax({
> > +          url: genotet.data.userURL,
> > +          type: 'POST',
> > +          data: userInfo,
> > +          dataType: 'json'
> > +        }).done(function(data) {
> > +            if (!data.success) {
>
> Why is this obsolete? This 'success' only indicates that the POST is
> successful.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/145/files/cecdab20766228a877f5032c6c96cafe153dde2b#r56754303>
>